### PR TITLE
Remove assert that SymbolDistinguisher descriptions are distinct

### LINKED ISSUE #8588: assert in symboldistinguisher.MakeDescriptions
here is callstack and dump location \\[share]\public\hechang\SymbolDump

you can do live repro by opening this (\\[share]\public\kevinh\SpuriousErroneus) with debug roslyn bits.

by looking at code, this assert seems might be wrong

                // The locations should not be equal, but they might be if the same
                // SyntaxTree is referenced by two different compilations.
                if (location0 == location1)
                {
                    var containingAssembly0 = unwrappedSymbol0.ContainingAssembly;
                    var containingAssembly1 = unwrappedSymbol1.ContainingAssembly;

                    // May not be the case if there are error types.
                    if ((object)containingAssembly0 != null && (object)containingAssembly1 != null)
                    {
                        location0 = containingAssembly0.Identity.ToString();
                        location1 = containingAssembly1.Identity.ToString();

                        // Even if the friendly locations produced by GetLocationString aren't
                        // distinct, the containing assembly identities should be.
                        Debug.Assert(location0 != location1);
                    }
                }

since if original symbols are metadata symbol that came from same assembly, location0 and 1 can be same.

>	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.SymbolDistinguisher.MakeDescriptions() Line 111	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.SymbolDistinguisher.GetDescription(int index) Line 195	C#
 	Microsoft.CodeAnalysis.CSharp.dll!Microsoft.CodeAnalysis.CSharp.SymbolDistinguisher.Description.ToString() Line 232	C#
 	mscorlib.dll!System.Text.StringBuilder.AppendFormatHelper(System.IFormatProvider provider, string format, System.ParamsArray args) Line 1466	C#
 	mscorlib.dll!string.FormatHelper(System.IFormatProvider provider, string format, System.ParamsArray args) Line 2963	C#
 	mscorlib.dll!string.Format(System.IFormatProvider provider, string format, object[] args) Line 2959	C#
 	Microsoft.CodeAnalysis.dll!Microsoft.CodeAnalysis.DiagnosticInfo.GetMessage(System.IFormatProvider formatProvider) Line 339	C#
 	Microsoft.CodeAnalysis.dll!Microsoft.CodeAnalysis.DiagnosticWithInfo.GetMessage(System.IFormatProvider formatProvider) Line 100	C#
 	Microsoft.CodeAnalysis.Workspaces.dll!Microsoft.CodeAnalysis.Diagnostics.DiagnosticData.Create(Microsoft.CodeAnalysis.Document document, Microsoft.CodeAnalysis.Diagnostic diagnostic) Line 413	C#
 	Microsoft.CodeAnalysis.Features.dll!Microsoft.CodeAnalysis.Diagnostics.EngineV1.DiagnosticIncrementalAnalyzer.GetDiagnosticData.AnonymousMethod__1(Microsoft.CodeAnalysis.Diagnostic d) Line 842	C#
 	System.Core.dll!System.Linq.Enumerable.WhereSelectEnumerableIterator<Microsoft.CodeAnalysis.Diagnostic, Microsoft.CodeAnalysis.Diagnostics.DiagnosticData>.MoveNext() Line 285	C#
 	System.Core.dll!System.Linq.Buffer<Microsoft.CodeAnalysis.Diagnostics.DiagnosticData>.Buffer(System.Collections.Generic.IEnumerable<Microsoft.CodeAnalysis.Diagnostics.DiagnosticData> source) Line 2558	C#
 	System.Core.dll!System.Linq.Enumerable.ToArray<Microsoft.CodeAnalysis.Diagnostics.DiagnosticData>(System.Collections.Generic.IEnumerable<Microsoft.CodeAnalysis.Diagnostics.DiagnosticData> source) Line 830	C#
 	System.Collections.Immutable.dll!System.Collections.Immutable.ImmutableArray.CreateRange<Microsoft.CodeAnalysis.Diagnostics.DiagnosticData>(System.Collections.Generic.IEnumerable<Microsoft.CodeAnalysis.Diagnostics.DiagnosticData> items) Line 137	C#
 	Microsoft.CodeAnalysis.Workspaces.dll!Microsoft.CodeAnalysis.ImmutableArrayExtensions.AsImmutableOrEmpty<Microsoft.CodeAnalysis.Diagnostics.DiagnosticData>(System.Collections.Generic.IEnumerable<Microsoft.CodeAnalysis.Diagnostics.DiagnosticData> items) Line 45	C#
 	Microsoft.CodeAnalysis.Features.dll!Microsoft.CodeAnalysis.Diagnostics.EngineV1.DiagnosticIncrementalAnalyzer.AnalyzerExecutor.GetDocumentAnalysisDataAsync(Microsoft.CodeAnalysis.Diagnostics.EngineV1.DiagnosticAnalyzerDriver analyzerDriver, Microsoft.CodeAnalysis.Diagnostics.EngineV1.DiagnosticIncrementalAnalyzer.StateSet stateSet, Microsoft.CodeAnalysis.Diagnostics.EngineV1.DiagnosticIncrementalAnalyzer.VersionArgument versions) Line 71	C#

### PR DESCRIPTION
**Pull request by cston**
Fixes #8588

### CODE REVIEW

**Comment by cston** [comment]
@dotnet/roslyn-compiler

File: src/Compilers/CSharp/Portable/Symbols/SymbolDistinguisher.cs [hunk 1]
```diff
@@ -103,12 +103,13 @@ private void MakeDescriptions()
...
+                        // assembly identities may be identical as well. (For instance, the
+                        // symbols are type arguments to the same generic type, and the type
+                        // arguments have the same string representation. The assembly
+                        // identities will refer to the generic types, not the type arguments.)
                         location0 = containingAssembly0.Identity.ToString();
                         location1 = containingAssembly1.Identity.ToString();
-
-                        // Even if the friendly locations produced by GetLocationString aren't
-                        // distinct, the containing assembly identities should be.
-                        Debug.Assert(location0 != location1);
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/Symbols/SymbolDistinguisher.cs]: It feels like augmenting description with the same location doesn't make diagnostics less confusing and the added test actually demonstrates this. I would say that we should leave descriptions as is in this case.

**Comment by cston** [comment]
@dotnet/roslyn-compiler please review.

File: src/Compilers/CSharp/Portable/Symbols/SymbolDistinguisher.cs (hunk 1 above)
Comment (cston) [inline src/Compilers/CSharp/Portable/Symbols/SymbolDistinguisher.cs]: Updated to not add locations if identical.

**Comment by AlekseyTs** [comment]
LGTM

**Pull request merged**